internal/peer/node/start.go

```diff
@@ -825,6 +825,7 @@ func serve(args []string) error {
 				serverEndorser,
 				discoveryService,
 				peerInstance,
+				&serverConfig.SecOpts,
```

Comment:
Just one small thing - here you passed the server config, not the client config. 
This would mean that you do not honor the `tls.clientKey` and `tls.clientCert`. 

We probably need to check if the client TLS config is defined, and if so, then use it instead and only otherwise fallback on the server TLS config.